apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/ws/WebSocketNetworkTransport.kt

```diff
@@ -173,9 +175,7 @@ private constructor(
             val webSocketConnection = try {
               webSocketEngine.open(
                   url = serverUrl,
-                  headers = mapOf(
-                      "Sec-WebSocket-Protocol" to protocolFactory.name,
-                  )
+                  headers = headers + HttpHeader("Sec-WebSocket-Protocol", protocolFactory.name),
```

Comment:
Should we only add the "Sec-WebSocket-Protocol" header if it's not present already? That'd allow callers to customize the header if really required? (not sure this will ever happen but it might?)